### PR TITLE
Support Dll for Windows [14019]

### PR DESCRIPTION
**Pull request by jparisu**
After #152 

This PR adds Dll support for Windows library generation

### CODE REVIEW

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: ddsrouter_core/cmake/test/cmake_test_utils.cmake [hunk 1]
```diff
@@ -129,7 +129,7 @@ function(add_test_executable TEST_EXECUTABLE_NAME TEST_SOURCES TEST_NAME TEST_LI
     )
 
     if(MSVC)
-        target_compile_definitions(${TEST_EXECUTABLE_NAME} PRIVATE _CRT_DECLARE_NONSTDC_NAMES=0)
+        target_compile_definitions(${TEST_EXECUTABLE_NAME} PRIVATE _CRT_DECLARE_NONSTDC_NAMES=0 ${SUBMODULE_PROJECT_MACROS}_SOURCE)
```
[the remark this answers is not in the document]
Comment (jparisu) [reply ddsrouter_core/cmake/test/cmake_test_utils.cmake:132]: all these files are the same file. Modifying .common is enough

File: tools/ddsrouter_tool/project_settings.cmake [hunk 2]
```diff
@@ -23,12 +23,16 @@ set(SUBMODULE_PROJECT_SUMMARY
...
+    yaml-cpp
+    fastcdr
+    fastrtps
     ddsrouter_utils
     ddsrouter_event
     ddsrouter_core
     ddsrouter_yaml)
 
 set(SUBMODULE_PROJECT_DEPENDENCIES
+    $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
```
Comment (rsanchez15) [inline tools/ddsrouter_tool/project_settings.cmake:35]: This is not required. Remove

File: tools/ddsrouter_tool/test/application/CMakeLists.txt [hunk 3]
```diff
@@ -77,6 +100,8 @@ if(PYTHONINTERP_FOUND)
 
     endif(WIN32)
 
+    message(STATUS "Test Environment: ${TEST_ENVIRONMENT}")
```
Comment (rsanchez15) [inline tools/ddsrouter_tool/test/application/CMakeLists.txt:103]: Remove debug trace

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — APPROVED** [review]
LGTM 🚀

**Pull request merged**